### PR TITLE
Add ambience switcher action plugin

### PR DESCRIPTION
**Pull request by Sagittarii**
Here is a first version of the ambience switcher plugin, based on a copy of the profile switcher code, as shown here : http://talk.maemo.org/showpost.php?p=1454090&postcount=30

Limitations so far are that the ambiances path/name are hardcoded in the user interface, but they can be changed manually in the rule.qml file if needed.

Feels free to let me know of any change needed so that it fits with the rest of phonebot's code.

Thanks !

### CODE REVIEW

**Comment by SfietKonstantin** [comment]
I'm not a huge fan of this hardcoded path inside phonebot. However, it looks nice, so I would like to accept it. What about accepting it, but removing the UI side at the moment (just have the action, that is accessible through QML file)

File: src/plugins/ambience/ambience.pro [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+HEADERS = ambienceaction.h
+
+SOURCES = plugin.cpp \
+    ambienceaction.cpp
+
+#include(../../3rdparty/libnemomw/ambience/ambience-include.pri)
+
+OTHER_FILES += \
+    notes.md
```
Comment (SfietKonstantin) [inline src/plugins/ambience/ambience.pro]: You forgot to git add this file

File: src/plugins/ambience/ambienceaction.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+ */
+
+#include "ambienceaction.h"
+#include <action_p.h>
+#include <choicemodel.h>
+#include <QtDBus/QDBusConnection>
+#include <QtDBus/QDBusInterface>
+
+static const char *AMBIANCE_KEY = "ambience";
+static const char *WORK_AMBIANCE_KEY = "file:///usr/share/ambience/storm/storm.ambience";
```
Comment (SfietKonstantin) [inline src/plugins/ambience/ambienceaction.cpp]: Remove or comment this at the moment. Let's not expose this plugin to the UI.

File: src/plugins/ambience/ambienceaction.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+QString AmbienceActionMeta::description() const
+{
+    return tr("This action allows to change the ambience when triggered.");
+}
+
+QString AmbienceActionMeta::summary(const QVariantMap &properties) const
+{
+    QString ambience = properties.value(AMBIANCE_KEY).toString();
+    if (ambience == WORK_AMBIANCE_KEY) {
```
Comment (SfietKonstantin) [inline src/plugins/ambience/ambienceaction.cpp]: Same, remove this at the moment.

File: src/plugins/ambience/ambienceaction.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+#include <action.h>
+#include <abstractmetadata.h>
+
+class AmbienceActionPrivate;
+class AmbienceAction : public Action
+{
+    Q_OBJECT
+    Q_PROPERTY(QString ambience READ ambience WRITE setAmbience NOTIFY ambienceChanged)
+    PHONEBOT_METADATA(AmbienceActionMeta)
```
Comment (SfietKonstantin) [inline src/plugins/ambience/ambienceaction.h]: Don't expose to the UI: remove this macro

**Comment by Sagittarii** [comment]
I have taken your requests into account. For now no more UI exposed, and I checked that the rules are still applied when the rule is written manually.
I remove the missing "notes" file from the project file, as it was only to help me locally during the development and not something to publish.

Here is an example rule for test, as you can not do it from GUI anymore:

[nemo@Jolla ~]$ cat .config/harbour-phonebot/harbour-phonebot/rule_00001/rule.qml 
import org.SfietKonstantin.phonebot.time 1.0
import org.SfietKonstantin.phonebot 1.0
import org.SfietKonstantin.phonebot.time 1.0
import org.SfietKonstantin.phonebot.ambience 1.0

Rule {
    actions: [
        AmbienceAction {
            ambience: "file:///usr/share/ambience/redtrain/redtrain.ambience"
        }
    ]
    condition: WeekDayCondition {
        onFriday: true
        onMonday: true
        onSaturday: true
        onSunday: true
        onThursday: true
        onTuesday: true
        onWednesday: true
    }
    mappers: [
        TimeMapper {
            id: mapper0
            hour: 23
            minute: 30
        }
    ]
    name: "Night"
    trigger: TimeTrigger {
        time: mapper0.value
    }
}

File: src/plugins/ambience/ambienceaction.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2014 Lucien XU <sfietkonstantin@free.fr>
```
Comment (SfietKonstantin) [inline src/plugins/ambience/ambienceaction.cpp]: I didn't wrote this file ! :)

File: src/plugins/ambience/plugin.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2014 Lucien XU <sfietkonstantin@free.fr>
```
Comment (SfietKonstantin) [inline src/plugins/ambience/plugin.cpp]: Same

**Comment by SfietKonstantin** [comment]
Can you squash your commits into one (and add in commit message that the UI is not here at the moment ?) 

thank you !

**Comment by Sagittarii** [comment]
Fixed copyright, and learnt that it was possible to squash commits with Git ;)

**Comment by SfietKonstantin** [comment]
LGTM, first pull request \o/

**Pull request merged**